src/portable/st/synopsys/dcd_synopsys.c

```diff
@@ -27,10 +27,24 @@
 
 #include "tusb_option.h"
 
-#if TUSB_OPT_DEVICE_ENABLED && CFG_TUSB_MCU == OPT_MCU_STM32F4
+#if TUSB_OPT_DEVICE_ENABLED && ( CFG_TUSB_MCU == OPT_MCU_STM32F4 || \
+                                 CFG_TUSB_MCU == OPT_MCU_STM32H7 )
+
+#if CFG_TUSB_MCU == OPT_MCU_STM32F4
+  #include "stm32f4xx.h"
+  // TODO Merge with OTG_HS
+  // Max endpoints for each direction
+  #define EP_MAX        USB_OTG_FS_MAX_IN_ENDPOINTS
+  #define EP_FIFO_SIZE  USB_OTG_FS_TOTAL_FIFO_SIZE
+#elif CFG_TUSB_MCU == OPT_MCU_STM32H7
+  #include "stm32h7xx.h"
+  // TODO There is no equivalent macro for max endpoints in H7 header.
+  #define EP_MAX        8
+  #define EP_FIFO_SIZE  4096
```

Comment:
For F1, use USB_OTG_FIFO_SIZE, but not sure about the endpoint count.

Reply:
`USB_OTG_FIFO_SIZE` has the correct value, but AFAICT it's not meant for actually querying how much FIFO space is present; it's used to get bounds on the amount of address space used for registers and I/O (look at the other variables it's grouped with in the header). So I'm hesitant to use that define for querying FIFO size.